packages/testing-library/utils/test.ts

```diff
@@ -44,15 +44,21 @@ async function elementExists(page: Page, selector: string) {
     return elementToCheck;
 }
 
-async function click(page: Page, selector: string, waitForNavigation: boolean) {
+/**
+ * @param path If the click action results in a navigation, provide the expected path
+ *
+ * @example
+ * ```js
+ * await click(page, "a.some-link", "/some-path");
+ * ```
+ */
+async function click(page: Page, selector: string, path?: string) {
     await elementExists(page, selector);
 
-    if (!waitForNavigation) {
-        await page.click(selector);
-    } else {
-        // if a click triggers a page reload, this is the correct
-        // syntax according to puppeteer documentation
-        await Promise.all([page.waitForNavigation(), page.click(selector)]);
+    await page.click(selector);
+
+    if (path) {
+        await page.waitForURL((url) => url.pathname === path);
```

Comment:
while `page.waitForNavigation()` _is_ an available api in playwright, it's deprecated. instead, it recommends to use `page.waitForURL`, which requires validation as to what we expect the url path to be, hence the change of `waitForNavigation` to `path`